### DISCUSSION TITLE
v1.9.1

### DISCUSSION THREAD

**Discussion by samuelcolvin** [pydantic/pydantic#3962] [General]
I'm going to try and get v1.9.1 out soon. Please comment here if you have an issue or pull request which needs to be merged or addressed.

@PrettyWood has already created [a milestone](https://github.com/samuelcolvin/pydantic/issues?q=is%3Aopen+is%3Aissue+milestone%3Av1.9.1), but I know most contributors can't assign to a milestone. Hence this discussion.

**Comment by nhtgl** [comment]
Today I came across issue that was described here https://github.com/samuelcolvin/pydantic/issues/2277. There's even an open PR addressing this https://github.com/samuelcolvin/pydantic/pull/3941. I would be grateful if you could review it.

**Comment by himbeles** [comment]
Hi. We have these open regression issues #3848, #3658, and this PR #3819 that fixes both of them. Thank you!

**Comment by aleksul** [comment]
Hi. There are many mentions of an annoying bug with Python 3.9 ([PEP585](https://peps.python.org/pep-0585/)) type hints (see issues #3879 #3297 #2798 [#1298 (comment)](https://github.com/samuelcolvin/pydantic/issues/1298#issuecomment-1068634254) [sqlmodel's #228](https://github.com/tiangolo/sqlmodel/issues/228) and possibly more). I've opened a PR (#3681) to fix this behavior and it will be nice to have it assigned to this milestone.

**Reply by samuelcolvin** [reply]
Thanks, added it to the milestone.

**Comment by 07pepa** [comment]
Hopefuly i am not too late.  
There is issue with serialization of time deltas with [timedelta_isoformat](https://github.com/samuelcolvin/pydantic/blob/d7a8272d7e0c151b0bd43df596be02e0d436ebdf/pydantic/json.py#L106).

it did not take into account negative time delta values
issue raised [here](https://github.com/samuelcolvin/pydantic/issues/3899).

TLDR:
a minus was in incorect place `P-1DT23H59M59.888735S` instead of correct `-P1DT23H59M59.888735S` as documented [here](https://pydantic-docs.helpmanual.io/usage/types/#datetime-types).

PR to fix this tiny error and add tests [here](https://github.com/samuelcolvin/pydantic/pull/3909).

it would be nice to fix this Asap and add it to milestone.

Thank your for your good work! (this bug was actualy discovered by pydantic itself so realy good work)

**Reply by samuelcolvin** [reply]
I'll make sure to check it out.

**Comment by oefe** [comment]
The new VS Code support has undesired (IMO) behavior when used with aliases: #3986 
This is otherwise a great feature, but when using aliases to map external identifiers (e.g. from a REST API) to nice pythonic names, I would want to use the pythonic identifiers to create objects, not the aliases.
Could this be addressed in v1.9.1?
(This is not an issue yet, just a discussion topic)

**Reply by samuelcolvin** [reply]
I suspect this well not be possible I think it's a fundamental limitation of PEP 681.

If you or anyone else can find a solution or work around, feel free to create an issue.
